packages/atomspace.nix

```diff
@@ -45,7 +45,13 @@ stdenv.mkDerivation rec {
     # prevent override of PYTHON_DEST
     sed -i -e 's/OUTPUT_VARIABLE PYTHON_DEST//g' $(find . -type f)
 
-    sed -i -e "s=/usr/local/share/opencog/scm=$out/share/opencog/scm=g" $(find . -type f)
+    sed -i -e "s=/usr/local/share/opencog/scm=${src}/opencog/scm=g" $(find . -type f)
+
+    ${import ../init-psql-db.nix} # prepare psql
+    createdb opencog_test # create test database
+    psql -c "CREATE USER opencog_tester WITH PASSWORD 'cheese';" # create test user
+    # NOTE: create with test user, or user will be nixbld and grants to other users seem to not work
+    cat ${src}/opencog/persist/sql/multi-driver/atom.sql | psql opencog_test -U opencog_tester
```

Comment:
This is correct, for the creation of the unit-test database.  More generally, the ability to grant is a permission itself ... requires that the creating user has "grant" permissions. Or something like that. I don't recall. So maybe you need this:
```
CREATE ROLE nixbld WITH CREATEDB;
ALTER ROLE nixbld WITH LOGIN;
```
or something like that -- whoever the final user is gets to create databases, and to login to use them.

Reply:
Or maybe `ALTER ROLE nixbld WITH CREATEUSER;` to create other users!?  Not quite sure what you are trying to do here.

Reply:
Nix creates `nixbld` user and its permissions so I'd rather not mess with it much. I did try several alter and grant queries but these seem to not stick around at all, grants are empty once I login with `opencog_tester`.